### PR TITLE
Fixing Logging Issues

### PR DESCRIPTION
**Pull request by rbharath**
This PR fixes the logging issues noted in #2023 and cleans up `data_loader.py` to be simpler and have more docs.

### CODE REVIEW

File: deepchem/data/tests/test_data_loader.py [hunk 1]
```diff
@@ -12,156 +9,153 @@
...
+def test_unlabelled():
+  current_dir = os.path.dirname(os.path.abspath(__file__))
+  input_file = os.path.join(current_dir, "../../data/tests/no_labels.csv")
+  featurizer = dc.feat.CircularFingerprint(size=1024)
+  loader = dc.data.CSVLoader(
+      tasks=[], smiles_field="smiles", featurizer=featurizer)
+  loader.create_dataset(input_file)
+
+
+def scaffold_test_train_valid_test_split():
```
Comment (peastman) [inline deepchem/data/tests/test_data_loader.py:21]: This won't get run because the name doesn't start with "test".  Same with a few others below.

Comment (rbharath) [reply deepchem/data/tests/test_data_loader.py]: Good point! Renamed

**Comment by rbharath** [comment]
I'm getting a lot of test failures on travis I'm not seeing on my local machine... Will try to see what's happening here

**Comment by rbharath** [comment]
I'm a little confused on this one since I'm not able to reproduce these errors locally. 

@peastman Do you have any thoughts on what could be causing a Travis/local-environment mismatch?

**Comment by rbharath** [comment]
I'm going to close this PR and try making a new one to see if that helps.

**Pull request closed without merging**